cli/src/cli_util.rs

```diff
@@ -2828,11 +2828,25 @@ impl CliRunner {
         let maybe_cwd_workspace_loader = WorkspaceLoader::init(find_workspace_dir(&cwd))
             .map_err(|err| map_workspace_load_error(err, None));
         layered_configs.read_user_config()?;
+        let mut repo_config_path = None;
         if let Ok(loader) = &maybe_cwd_workspace_loader {
             layered_configs.read_repo_config(loader.repo_path())?;
+            repo_config_path = Some(layered_configs.repo_config_path(loader.repo_path()));
         }
         let config = layered_configs.merge();
-        ui.reset(&config)?;
+        if let Err(e) = ui.reset(&config) {
+            let user_config_path = layered_configs.user_config_path()?;
+            let paths = [repo_config_path, user_config_path]
+                .into_iter()
+                .filter_map(|path| path.map(|x| format!("- {}", x.display())))
+                .join("\n");
+            writeln!(
+                ui.warning_default(),
+                "{}\nCheck the following config files:\n{}",
+                e.error,
+                paths
+            )?;
+        }
```

Comment:
Better to return the error here (instead of relying on the next `ui.reset()`.)

Perhaps, the warning can be attached as a hint. `map_err(|e| e.hinted(format!(..)))?`

Reply:
Oh, didn't know about that one. That makes sense.